agent/api/container.go

```diff
@@ -62,47 +62,54 @@ type Container struct {
 	DockerConfig           DockerConfig                `json:"dockerConfig"`
 	RegistryAuthentication *RegistryAuthenticationData `json:"registryAuthentication"`
 
-	// DesiredStatus represents the state where the container should go.  Generally the desired status is informed by the
-	// ECS backend as a result of either API calls made to ECS or decisions made by the ECS service scheduler, though the
-	// agent may also set the DesiredStatus if a different "essential" container in the task exits.  The DesiredStatus is
-	// almost always either ContainerRunning or ContainerStopped.  Do not access DesiredStatus directly.  Instead,
-	// use `GetDesiredStatus` and `SetDesiredStatus`.
-	// TODO DesiredStatus should probably be private with appropriately written setter/getter.  When this is done, we need
-	// to ensure that the UnmarshalJSON is handled properly so that the state storage continues to work.
+	// DesiredStatus represents the state where the container should go. Generally,
+	// the desired status is informed by the ECS backend as a result of either
+	// API calls made to ECS or decisions made by the ECS service scheduler,
+	// though the agent may also set the DesiredStatus if a different "essential"
+	// container in the task exits. The DesiredStatus is almost always either
+	// ContainerRunning or ContainerStopped.
+	// NOTE: Do not access DesiredStatus directly.  Instead, use `GetDesiredStatus`
```

Comment:
I haven't gotten to `containers.go` yet. Working my way through `task.go`. Will change these when I get there. I'll add a comment for when I think this PR is ready to be reviewed again.